### ISSUE TITLE
[no-floating-promises] doesn't catch floating promises in if blocks

### ISSUE THREAD

**Issue by gordonmleigh** [typescript-eslint/typescript-eslint#1676]
**Repro**

```JSON
{
  "parser": "@typescript-eslint/parser",
  "extends": ["plugin:@typescript-eslint/recommended"],
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module",
    "project": "tsconfig.json"
  },
  "rules": {
    "@typescript-eslint/no-floating-promises": "error"
  }
}
```

```TS
async function decide() {
  return new Promise(resolve => resolve());
}

async function main() {
  if (!decide()) { // all good
    return;
  }
  decide(); // error
}
```

**Expected Result**

I'd expect the first call to `decide` to give an error, since the promise isn't handled.

**Actual Result**

Only the second call to `decide` causes an error.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.11.1` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
I disagree.
This rule is not about having a promise just sitting as a lonely expression, it's about ensuring you are correctly handling the promise.

`if (promise) {}` does not handle the promise at all, nor does it use the promise correctly.

**Comment by bradzacher** [comment]
This is case is handled by our rule `no-misused-promises`.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-promises.md

It won't be added to this rule for that reason, and also because this rule is mainly intended to help catch places you forgot to `await` a promise.
